better name for geotoh3_lon_lat_input_order

bharatnc opened the pull request:
Follow up for https://github.com/ClickHouse/ClickHouse/pull/78852. Rename `geotoh3_lon_lat_input_order` to `geotoh3_argument_order` which accepts `'lat_lon'` or `'lon_lat'` for values.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

bharatnc committed:
better name for geotoh3_lon_lat_input_order

alexey-milovidov commented:
@bharatnc Let's rename to `geotoh3_argument_order` with enum values `lat_lon` and `lon_lat`.

bharatnc committed:
rename to geotoh3_argument_order and use enum values

bharatnc commented:
> @bharatnc Let's rename to `geotoh3_argument_order` with enum values `lat_lon` and `lon_lat`.

sure, changed.

bharatnc committed:
fix to use enum type

bharatnc committed:
fix test

bharatnc commented:
> > sure, changed.
> 
> It should be an enum, not a string. Then you don't need to do any validation yourself and the code is much cleaner

Sure, fixed.

bharatnc commented:
Test failures not related: 

[Integration tests (aarch64, distributed plan, 4/4)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=80544&sha=766d77bddfeb45f81e7a6145d2907bfc268e8d19&name_0=PR&name_1=Integration%20tests%20%28aarch64%2C%20distributed%20plan%2C%204%2F4%29) - `test_database_glue`
